crates/data_components/src/postgres.rs

```diff
@@ -361,10 +407,19 @@ impl Postgres {
         row.get(0)
     }
 
-    async fn insert_batch(&self, transaction: &Transaction<'_>, batch: RecordBatch) -> Result<()> {
+    async fn insert_batch(
+        &self,
+        transaction: &Transaction<'_>,
+        batch: RecordBatch,
+        on_conflict: Option<OnConflict>,
+    ) -> Result<()> {
         let insert_table_builder = InsertBuilder::new(&self.table_name, vec![batch]);
+
+        let sea_query_on_conflict =
```

Comment:
Here for Postgres we translate our `OnConflict` struct into the `sea_query::OnConflict` struct and pass it to the builder.